@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,182 @@ export namespace MapViewUtils {
-        const tiltSign = Math.sign(up.dot(mapTargetNormal));
+        const clampedDeltaTilt = computeClampedDeltaTilt(
+            mapView,
+            offsetX,
+            offsetY,
+            deltaTilt,
+            maxTiltAngle,
+            mapTargetWorld,
+            rotationTargetWorld
+        );
 
-        const currentTilt = (tiltSign === 0 ? 1 : tiltSign) * dir.angleTo(mapTargetNormal);
+        applyTiltAroundTarget(mapView, rotationTargetWorld, clampedDeltaTilt);
+    }
 
-        //FIXME(HARP-11926): For globe tilt in the map center is different from the tilt in the rotation center,
-        //hence the clamped tilt is too conservative.
-        const clampedDeltaTilt =
-            MathUtils.clamp(deltaTilt + currentTilt, 0, maxTiltAngle) - currentTilt;
-        if (Math.abs(clampedDeltaTilt) <= Number.EPSILON) {
-            return;
+    /**
+     * @hidden
+     * @internal
+     *
+     * Applies the given Azimith to the camera around the supplied target.
+     */
+    function applyAzimuthAroundTarget(
+        mapView: MapView,
+        rotationTargetWorld: THREE.Vector3,
+        deltaAzimuth: number
+    ) {
+        const camera = mapView.camera;
+        const projection = mapView.projection;
+
+        const headingAxis = projection.surfaceNormal(rotationTargetWorld, cache.vector3[0]);
+        const headingQuat = cache.quaternions[0].setFromAxisAngle(headingAxis, deltaAzimuth);
+        camera.quaternion.premultiply(headingQuat);
+        camera.position.sub(rotationTargetWorld);
+        camera.position.applyQuaternion(headingQuat);
+        camera.position.add(rotationTargetWorld);
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Clamps the supplied `deltaTilt` to the `maxTiltAngle` supplied. Note, when a non-zero offset
+     * is applied, we apply another max angle of 89 degrees to the rotation center to prevent some
+     * corner cases where the angle at the rotation center is 90 degrees and therefore intersects
+     * the geometry with the near plane.
+     */
+    function computeClampedDeltaTilt(
+        mapView: MapView,
+        offsetX: number,
+        offsetY: number,
+        deltaTilt: number,
+        maxTiltAngle: number,
+        mapTargetWorld: THREE.Vector3,
+        rotationTargetWorld: THREE.Vector3
+    ): number {
+        const camera = mapView.camera;
+        const projection = mapView.projection;
+
+        const tilt = extractTiltAngleFromLocation(projection, camera, mapTargetWorld);
+        if (tilt + deltaTilt < 0) {
+            // Clamp the final tilt to 0
+            return -tilt;
+        } else if (deltaTilt <= 0) {
+            // Reducing the tilt isn't clamped (apart from above).
+            return deltaTilt;
+        }
+
+        const rotationCenterTilt =
+            offsetX === 0 && offsetX === offsetY
+                ? tilt
+                : extractTiltAngleFromLocation(projection, camera, rotationTargetWorld!);
```

Comment:
if the rotation and map targets are the same point, most of the code after this is not needed anymore, i.e., there's no triangle anymore. We only need to check the tilt on the map target.

Reply:
Good point, thanks.